tensorflow/core/platform/s3/s3_file_system.cc

```diff
@@ -271,7 +278,7 @@ class S3WritableFile : public WritableFile {
         transfer_manager_(transfer_manager),
         sync_needed_(true),
         outfile_(Aws::MakeShared<Aws::Utils::TempFile>(
-            kS3FileSystemAllocationTag, "/tmp/s3_filesystem_XXXXXX",
+            kS3FileSystemAllocationTag, kS3TempFileTemplate,
```

Comment:
I think it would be better to get the temp directory from the environment (Env::GetTempDirectories or environment variables) but that's optional for now and can be done in a later PR.

Reply:
I think we should not invest too much time on this detail, as we will switch to `s3_filesystem_plugins` soon

Reply:
Totally agree. Thank you